### PR TITLE
r-devtools: new version and dependencies

### PR DESCRIPTION
**Pull request by ajkotobi**

### CODE REVIEW

File: var/spack/repos/builtin/packages/r-devtools/package.py [hunk 1]
```diff
@@ -13,27 +13,50 @@ class RDevtools(RPackage):
...
     depends_on('r-rcmdcheck@1.3.3:', when='@2.0.0:', type=('build', 'run'))
     depends_on('r-remotes@2.1.0:', when='@2.0.0:', type=('build', 'run'))
+    depends_on('r-remotes@2.1.1:', when='@2.3.0:', type=('build', 'run'))
+    depends_on('r-rlang@0.4.5:', when='@2.3.0:', type=('build', 'run'))
     depends_on('r-roxygen2@6.1.1:', when='@2.0.0:', type=('build', 'run'))
+    depends_on('r-roxygen2@7.1.0:', when='@2.3.0:', type=('build', 'run'))
     depends_on('r-sessioninfo@1.1.1:', when='@2.0.0:', type=('build', 'run'))
     depends_on('r-testthat@2.1.1:', when='@2.0.0:', type=('build', 'run'))
+    depends_on('r-testthat@2.3.2:', when='@2.3.0:', type=('build', 'run'))
+    depends_on('r-rversions@2.0.1:', when='@2.3.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-devtools/package.py]: Could you sort these dependencies in the same order they appear on https://cran.r-project.org/web/packages/devtools/index.html? That would make my life as a reviewer much easier 😄

Comment (ajkotobi) [reply var/spack/repos/builtin/packages/r-devtools/package.py]: Sure, it's re-ordered ;)

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/r-devtools/package.py [hunk 2]
```diff
@@ -13,27 +13,50 @@ class RDevtools(RPackage):
...
     depends_on('r-callr', when='@2.0.0:', type=('build', 'run'))
+    depends_on('r-callr@3.4.3:', when='@2.3.0:', type=('build', 'run'))
     depends_on('r-cli', when='@2.0.0:', type=('build', 'run'))
+    depends_on('r-cli@2.0.2:', when='@2.3.0:', type=('build', 'run'))
+    depends_on('r-covr@3.5.0:', when='@2.3.0:', type=('build', 'run'))
+    depends_on('r-crayon@1.3.4:', when='@2.3.0:', type=('build', 'run'))
+    depends_on('r-desc@1.2.0:', when='@2.3.0:', type=('build', 'run'))
+    depends_on('r-dt@0.13:', when='@2.3.0:', type=('build', 'run'))
+    depends_on('r-digest', type=('build', 'run'))
+    depends_on('r-digest@0.6.25', when='@3.2.0:' type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/r-devtools/package.py]:
```suggestion
    depends_on('r-digest@0.6.25:', when='@3.2.0:', type=('build', 'run'))
```
This syntax error is causing the unit tests to fail.

Comment (ajkotobi) [reply var/spack/repos/builtin/packages/r-devtools/package.py]: Fixed, thanks

**Comment by scheibelp** [comment]
Close/reopen to restart tests

**Pull request merged**